src/libraries/System.IO.FileSystem/tests/Enumeration/AttributeTests.cs

```diff
@@ -113,18 +113,41 @@ public void DirectoryAttributesAreExpected()
         }
 
         [Fact]
-        public void IsHiddenAttribute()
+        [PlatformSpecific(TestPlatforms.Windows | TestPlatforms.OSX)]
+        public void IsHiddenAttribute_Windows_OSX()
         {
-            DirectoryInfo testDirectory = Directory.CreateDirectory(GetTestFilePath());
-            FileInfo fileOne = new FileInfo(Path.Combine(testDirectory.FullName, GetTestFileName()));
+            // Windows and MacOS hide a file by setting the hidden attribute
+            IsHiddenAttributeInternal(useDotPrefix: false, useHiddenFlag: true);
+        }
 
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.AnyUnix)]
+        public void IsHiddenAttribute_Unix()
+        {
             // Put a period in front to make it hidden on Unix
-            FileInfo fileTwo = new FileInfo(Path.Combine(testDirectory.FullName, "." + GetTestFileName()));
+            IsHiddenAttributeInternal(useDotPrefix: true, useHiddenFlag: false);
+        }
+
+        private void IsHiddenAttributeInternal(bool useDotPrefix, bool useHiddenFlag)
+        {
+            string prefix = useDotPrefix ? "." : "";
+
+            DirectoryInfo testDirectory = Directory.CreateDirectory(GetTestFilePath());
+
+            FileInfo fileOne = new FileInfo(Path.Combine(testDirectory.FullName, GetTestFileName()));
+            FileInfo fileTwo = new FileInfo(Path.Combine(testDirectory.FullName, prefix + GetTestFileName()));
```

Comment:
Is it intentional to use Path.Combine() instead of Path.Join()?

Reply:
Yes (I prefer making sure the path gets rooted), but it also it doesn't matter much since these are unit tests.

Reply:
Yes (I prefer making sure the path gets rooted), but it also it doesn't matter much since these are unit tests.